### PR TITLE
Use the v0 provider and switch to using the claims parameter

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
# Summary

Updates this project to utilize the new v0 provider. Highlights:

- The Banno scope is not used. Custom claims must be requested via the OAuth claims parameter.
- PKCE is enabled
- Discovery is utilized to configure the client

# Jira Ticket

[DX-437 Update `Consumer API OIDC Example` to use V0 auth endpoints](https://banno-jha.atlassian.net/browse/DX-437)

### CODE REVIEW

**Comment by ChadKillingsworth** [comment]
Items still not working:
 - Consent won't function correctly until https://github.com/Banno/platform-ux/pull/9890 is merged and deployed.
 - Some of the claims cannot be requested individually until https://github.com/Banno/node-openid-provider/pull/419 is merged and deployed

**Comment by ChadKillingsworth** [comment]
Claims may now be individually requested. I pushed changes to utilize the claims parameter rather than the special OIDC scopes.

Screen Shot 2020-12-15 at 5 47 59 AM

File: server.js [hunk 1]
```diff
@@ -16,20 +16,21 @@
...
+const session = require('express-session');
+const config = require('./config');
+
+const env = process.env.ENVIRONMENT;
+console.log(`Environment: ${env}`);
+
+(async () => {
 // Configure the OpenID Connect client based on the issuer.
-const issuer = new Issuer(config.issuer[`garden-${env}`]);
+const issuer = await Issuer.discover(config.issuer[`garden-${env}`]);
```
Comment (DevWithTheHair) [inline server.js:33]: **praise:** I was able to run this and get the issuer that I expected (_https://digital.garden-fi.com/a/consumer/api/v0/oidc_) for this FI and the claims (including _https://api.banno.com/consumer/claim/institution_id_) that I expected for my user. 👍

File: server.js [hunk 2]
```diff
@@ -39,14 +40,28 @@ client[custom.clock_tolerance] = 300; // to allow a 5 minute clock skew for veri
 // This method is NOT recommended for use in production systems.
 let accessToken;
 
+const claims = {
```
Comment (DevWithTheHair) [inline server.js:43]: **suggestion:** We may want to include a comment which explains that these are just _some_ of the possible `claims` which can be requested.

This suggestion is analogous to what we do for the `scope` parameter which has this comment to help folks understand what they are reading:

```javascript
// These are the OpenID Connect scopes that you'll need.
```

**Review by DevWithTheHair — APPROVED** [review]
**praise:** Looks good to me (with non-blocking feedback). 👍

**Pull request merged**